src/Chainweb/Pact/PactService.hs

```diff
@@ -418,40 +422,80 @@ finalizeCheckpointer finalize = do
 _liftCPErr :: Either String a -> PactServiceM cas a
 _liftCPErr = either internalError' return
 
+type Balances  = HM.HashMap Text Decimal
+type GasTotals = HM.HashMap Text Decimal
+
+-- | The principal validation logic for groups of Pact Transactions.
+--
+-- Skips validation for genesis transactions, since gas accounts, etc. don't
+-- exist yet.
+--
 validateChainwebTxs
     :: PactDbEnv'
     -> Checkpointer
     -> BlockCreationTime
     -> BlockHeight
     -> Vector ChainwebTransaction
     -> IO (Vector Bool)
-validateChainwebTxs dbEnv cp blockOriginationTime bh txs =
-    V.mapM checkOne txs
+validateChainwebTxs dbEnv cp blockOriginationTime bh txs
+    | bh == 0 = pure $ V.replicate (V.length txs) False
```

Comment:
Previous code had `True` here -- does this work? It might prevent genesis txs from validating. Change `$` to `$!`

Reply:
You're right, I got that backwards.